### PR TITLE
Fix getting shares in a folder on Oracle

### PR DESCRIPTION
**Pull request by CarlSchwan**
Reported here https://github.com/nextcloud/server/issues/28960#issuecomment-1110912934

CC @nicro87

### CODE REVIEW

File: lib/private/Share20/DefaultShareProvider.php [hunk 1]
```diff
@@ -671,29 +671,43 @@ public function getSharesInFolder($userId, Folder $node, $reshares) {
...
 
 		$qb->orderBy('id');
 
-		$cursor = $qb->execute();
 		$shares = [];
-		while ($data = $cursor->fetch()) {
-			$shares[$data['fileid']][] = $this->createShare($data);
+
+		// make sure we execute the query even if there is no $childMountRootIds
+		if (count($childMountRootIds) === 0) {
```
Comment (CarlSchwan) [inline lib/private/Share20/DefaultShareProvider.php:694]: maybe could be replaced by a do while loop, not sure that is the more elegant here ...

**Comment by nicro87** [comment]
carry forward from comment https://github.com/nextcloud/server/issues/28960#issuecomment-1110912934
---

Hi @CarlSchwan

I tried your fix and it works for me. All errors are gone.

Thanks again for the quick help,
Nico

File: lib/private/Share20/DefaultShareProvider.php (hunk 1 above)
Comment (come-nc) [reply lib/private/Share20/DefaultShareProvider.php:694]: What about:
```
if (count($childMountRootIds) === 0) {
  $chunks = [[]];
} else {
  $chunks = array_chunk($childMountRootIds, 1000);
}
```

And then the same foreach as below?

**Review by come-nc — CHANGES_REQUESTED** [review]

**Review by come-nc — APPROVED** [review]

File: lib/private/Share20/DefaultShareProvider.php [hunk 2]
```diff
@@ -671,29 +671,39 @@ public function getSharesInFolder($userId, Folder $node, $reshares) {
...
-		$cursor = $qb->execute();
 		$shares = [];
-		while ($data = $cursor->fetch()) {
-			$shares[$data['fileid']][] = $this->createShare($data);
+
+		if (count($childMountRootIds) === 0) {
+			$chunks = [[]];
+		} else {
+			$chunks = array_chunk($childMountRootIds, 1000);
+		}
```
Comment (artonge) [inline lib/private/Share20/DefaultShareProvider.php:697]: Don't know, might be more readable :)

```suggestion
		$chunks = array_chunk($childMountRootIds, 1000);

		// Force the request to be run when there is 0 mount.
		if (count($chunks) === 0) {
			$chunks = [[]];
		}
```

**Review by artonge — APPROVED** [review]

**Pull request merged**